polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1044,8 +1142,18 @@ MLIRScanner::EmitSYCLOps(const clang::Expr *Expr,
       }
 
       std::string name = MLIRScanner::getMangledFuncName(*Func, Glob.getCGM());
-      Op = builder.create<mlir::sycl::SYCLCallOp>(
-          loc, OptRetType, OptFuncType, Func->getNameAsString(), name, Args);
+      if (OptFuncType) {
```

Comment:
Line 1129 check that `OptFuncType` isn't NULL. Perhaps that check should be changed into an assert and here you do not have to check whether `OptFuncType` is not NULL.

Reply:
I think we don't want to do that yet because non-member functions in the `sycl` namespace will still go through that pass and fail the assert, e.g., `sycl::abs`

Reply:
Ok make sense then (for now).